pkg/steps/multi_stage/gen.go

```diff
@@ -163,6 +163,13 @@ func (s *multiStageTestStep) generatePods(
 				pod.Spec.DNSPolicy = coreapi.DNSNone
 			}
 		}
+		if step.NodeArchitecture != "" {
+			if pod.Spec.NodeSelector == nil {
+				// if nodeArchitecture is not set, default to amd64 node selector
+				pod.Spec.NodeSelector = map[string]string{"kubernetes.io/arch": string(api.NodeArchitectureAMD64)}
+			}
```

Comment:
What is the point of this block? It looks like it is always replaced on line 171

Reply:
missed the else 🤦🏻